app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ScanFragment.java

```diff
@@ -0,0 +1,181 @@
+package com.example.healthinspector.Fragments.ScanFlow;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentTransaction;
+
+import android.os.Parcelable;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.budiyev.android.codescanner.CodeScanner;
+import com.budiyev.android.codescanner.CodeScannerView;
+import com.budiyev.android.codescanner.DecodeCallback;
+import com.example.healthinspector.Models.ScannedProduct;
+import com.example.healthinspector.R;
+import com.example.healthinspector.databinding.FragmentScanBinding;
+import com.example.healthinspector.databinding.FragmentUserProfileBinding;
+import com.google.zxing.Result;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.parceler.Parcels;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+
+
+public class ScanFragment extends Fragment {
+
+    private FragmentScanBinding binding;
+    private CodeScanner codeScannerView;
+    private static final String TAG = "ScanFragment";
+    private HashMap<Integer, String> novaGroups = new HashMap<>();
```

Comment:
Rename to novaGroupsMap cause it's easy to get confused between novaGroup's' and novaGroup